apps/shared/lib/gbfs_metadata.ex

```diff
@@ -174,6 +175,72 @@ defmodule Transport.Shared.GBFSMetadata do
     end
   end
 
+  def stats(%{"data" => _data} = payload) do
+    stations_statistics(payload)
+    |> Map.merge(vehicle_statistics(payload))
+    |> Map.merge(%{version: 1})
+  end
+
+  def stations_statistics(%{"data" => _data} = payload) do
+    feed_url = payload |> first_feed() |> feed_url_by_name(:station_status)
+
+    with {:feed_exists, true} <- {:feed_exists, not is_nil(feed_url)},
+         {:ok, %HTTPoison.Response{status_code: 200, body: body}} <- http_client().get(feed_url),
+         {:ok, json} <- Jason.decode(body) do
+      stations = json["data"]["stations"]
+
+      %{
+        nb_stations: Enum.count(stations),
+        nb_installed_stations: Enum.count(stations, & &1["is_installed"]),
+        nb_renting_stations: Enum.count(stations, & &1["is_renting"]),
+        nb_returning_stations: Enum.count(stations, & &1["is_returning"]),
+        nb_docks_available: stations |> Enum.map(& &1["num_docks_available"]) |> non_nil_sum(),
+        nb_docks_disabled: stations |> Enum.map(& &1["num_docks_disabled"]) |> non_nil_sum(),
+        nb_vehicles_available_stations: stations |> Enum.map(&vehicles_available/1) |> non_nil_sum(),
+        nb_vehicles_disabled_stations: stations |> Enum.map(& &1["num_vehicles_disabled"]) |> non_nil_sum()
+      }
+    else
+      {:feed_exists, false} ->
+        %{}
+
+      e ->
+        Logger.error("Cannot get GBFS station_status details: #{inspect(e)}")
+        %{}
+    end
+  end
+
+  def vehicle_statistics(%{"data" => _data} = payload) do
+    feed_url = payload |> first_feed() |> feed_url_by_name(:vehicle_status)
+
+    with {:feed_exists, true} <- {:feed_exists, not is_nil(feed_url)},
+         {:ok, %HTTPoison.Response{status_code: 200, body: body}} <- http_client().get(feed_url),
+         {:ok, json} <- Jason.decode(body) do
+      vehicles = json["data"]["vehicles"] || json["data"]["bikes"]
+      nb_vehicles = Enum.count(vehicles)
+      nb_docked_vehicles = Enum.count(vehicles, &Map.has_key?(&1, "station_id"))
+
+      %{
+        nb_vehicles: nb_vehicles,
+        nb_disabled_vehicles: Enum.count(vehicles, & &1["is_disabled"]),
+        nb_reserved_vehicles: Enum.count(vehicles, & &1["is_reserved"]),
+        nb_docked_vehicles: nb_docked_vehicles,
+        nb_freefloating_vehicles: nb_vehicles - nb_docked_vehicles
```

Comment:
Je ne sais pas si c’est pertinent de faire ce calcul dans le code ? Toutes les autres stats sont des éléments directement renvoyés par le feed (avec juste un count éventuel) sauf celui-là, donc on pourrait croire que que c’est renvoyé par le feed.

En plus, est-ce que nb_vehicules ça compte les véhicules à disposition ou tous les véhicules, y compris en circulation ? La différence entre nb_vehicules et nb_docked_vehicules peut soit vouloir dire qu’il y a des vélos en freefloating à disposition (sans location en cours), soit qu’il y a des vélos en cours de location.

Reply:
> donc on pourrait croire que que c’est renvoyé par le feed.

C'est bien le cas, c'est pour éviter de traverser une nouvelle fois le feed. Sinon ce serait l'inverse de `nb_docked_vehicles`.

Soit `Enum.count(vehicles, & not Map.has_key?(&1, "station_id"))`.

Les différentes valeurs correspondent à la situation où les véhicules ne sont pas en location.

https://gbfs.org/specification/reference/#vehicle_statusjson

> Describes all vehicles that are not currently in active rental.